File: src/test/java/no/utgdev/diy/ex3/Ex3Test.java
package no.utgdev.diy.ex3;

import no.utgdev.diy.DIYStatic;
import org.junit.jupiter.api.Test;

import java.lang.reflect.Method;
import java.util.Map;
import java.util.Set;

import static org.assertj.core.api.Assertions.assertThat;

class Ex3Test {
    @Test
    void wireup() {
        Set<Method> methods = DIYStatic.scan(Ex3Config.class);
        Map<String, Object> objects = DIYStatic.instansiate(methods);
        DIYStatic.wireup(objects);
        Controller controller = (Controller) DIYStatic.getBean(objects, "controller");

        assertThat(controller).isNotNull();
        assertThat(controller.helloWorld()).isEqualTo("Hello, World");
    }
}

File: src/test/kotlin/no/utgdev/diy/ex3/Ex3Test.kt
package no.utgdev.diy.ex3

import no.utgdev.diy.DIYStatic.instansiate
import no.utgdev.diy.DIYStatic.wireup
import no.utgdev.diy.DIYStatic.getBean
import org.junit.jupiter.api.Test
import no.utgdev.diy.DIYStatic
import org.assertj.core.api.Assertions
import java.lang.reflect.Method

internal class Ex3Test {
    @Test
    fun wireup() {
        val methods: Set<Method> = DIYStatic.scan(Ex3Config::class.java)
        val objects: Map<String, Any?> = instansiate(methods)
        wireup(objects)
        val controller = getBean(objects, "controller") as Controller?
        Assertions.assertThat(controller).isNotNull
        Assertions.assertThat(controller!!.helloWorld()).isEqualTo("Hello, World")
    }
}